src/emc/pythonplugin/python_plugin.cc

```diff
@@ -36,8 +36,12 @@ namespace bp = boost::python;
 #define ERRMSG(fmt, args...)					\
     do {							\
         char msgbuf[MAX_ERRMSG_SIZE];				\
-        snprintf(msgbuf, sizeof(msgbuf) -1,  fmt, ##args);	\
-        error_msg = std::string(msgbuf);			\
+        size_t ret = snprintf(msgbuf, sizeof(msgbuf) -1,  fmt, ##args); \
+        if (ret > sizeof(msgbuf)){                                      \
```

Comment:
Should be `ret >= sizeof(msgbuf)`  The returned value doesn't count the '\0', so if you have buf[4] and write 'abcd' to it, snprintf will truncate to 'abc\0' and return 4.

Reply:
I was wondering about that, but decided to follow examples I found on the internet